addon/mixins/controllers/saving.js

```diff
@@ -48,10 +48,16 @@ export default Ember.Mixin.create(
 
     /* If there is a custom validations method, resolve it */
 
-    if (runCustomValidations && !runCustomValidations.then) {
-      Ember.assert('runCustomValidations() must return a promise (e.g. return new Ember.RSVP.Promise()).');
-    } else if (runCustomValidations) {
-      runCustomValidations().then(resolve, reject);
+    if (this.runCustomValidations) {
+      customValidationsPromise = this.runCustomValidations();
```

Comment:
if done the previous way (i.e.
```
var runCustomValidations = this.customValidations;
...

customValidationsPromise = runCustomValidations();
```

then  ```this === window```. An alternative way to accomplish what I did here is:

```
var runCustomValidations = this.customValidations;
...

customValidationsPromise = runCustomValidations.apply(this);
```